### ISSUE TITLE
Request a way that pass data from ModelLoader to ResourceDecoder.

### ISSUE THREAD

**Issue by peerless2012** [bumptech/glide#4966]
**Glide Version**: 
4.14.2

**Integration libraries**:
com.github.bumptech.glide:glide4.14.2
com.github.bumptech.glide:okhttp3-integration:4.14.2
com.github.bumptech.glide:compiler:4.14.2

**Device/Android Version**:
Pixel 5, Android 13

**Issue details / Repro steps / Use case background**: 
I use custom ModelLoader to load custom entry, and the entry have a zip file to load,.
The zip may from android assets, uri, or url, and i don't want to load this file and cache it by myself, but want to use glide loader to load it to InputStream  and decode it.
And when i decode the zip InputStream, the zip has a password, and the ResourceDecoder input is a InputStream，there is no way to get the password.
So I create this issue to find out , if there is a way to pass data to StreamDecoder?

**Comment by sjudd** [comment]
ModelLoader's accept any arbitrary Model, so you could pass the path to the zip and the password in the model directly. The ModelLoader can then load the zip file and convert it to an InputStream. At that point, presumably the ModelLoader could also wrap the zip stream in a second stream that uses the password to decode the data?

Providing decrypted data from the ModelLoader is the most robust way to deal with this in Glide because it does not require modifying all ResourceDecoders to be aware of your encrypted data.

**Comment by peerless2012** [comment]
But if my zip is from remote or assets or a content provider, if I build the inputstream in the modelloader, I have to fetch the stream from remote, and save it on disk, and if the file from assets, i have to impl the assets loader, and so on. That's very complexity.

**Comment by sjudd** [comment]
Do you know it's a ZIP file before you start the image load? If so you can create a custom model that contains the Uri and decryption data. Your custom model loader can delegate to the existing ModelLoader to fetch the data, but wrap the returned DataFetcher to capture the InputStream result and decrypt it.

StringLoader, as an example, delegates to the UriLoader: https://github.com/bumptech/glide/blob/18bba927a5e5fe7d07ada9667e9e503b9f0596a2/library/src/main/java/com/bumptech/glide/load/model/StringLoader.java#L35.

You'd need to add on to that additionally by wrapping the result of `uriLoader.buildLoadData(...)` in a new `DataFetcher` that intercepts the callback.

**Comment by peerless2012** [comment]
Yes, I build the zip url/uri and psw in a model, and custom a ModelLoader, in the ModelLoader, i get the uri from the model, and pass it to a delegate ModelLoader as [string,InputStream], but i can't do unzip operation before the delegate ModelLoader success load the url/uri (if the url is remote, i hope glide can read it and cache it, not by myself), when the delegate success, it will go to ResourceDecoder, and I have noway to pass the psw to the ResourceDecoder.
